File: frontend/src/app/shared/models/cart.model.ts
export interface CartItem {
  id: string;
  productVariantId: string;
  productName: string;
  variantInfo: string;
  price: number;
  quantity: number;
  imageUrl: string | null;
}

export interface Cart {
  id: string;
  userId: string | null;
  sessionId: string | null;
  items: CartItem[];
  totalAmount: number;
}

export interface AddToCartRequest {
  productVariantId: string;
  quantity: number;
}

File: backend/src/main/kotlin/com/tileforms/cart/CartModel.kt
package com.tileforms.cart

import java.math.BigDecimal
import java.util.UUID

data class CartItemModel(
    val id: String = UUID.randomUUID().toString(),
    val productVariantId: UUID,
    val productName: String,
    val variantInfo: String,
    val price: BigDecimal,
    val quantity: Int,
    val imageUrl: String?
)

data class CartModel(
    val id: String,
    val userId: String?,
    val sessionId: String?,
    val items: List<CartItemModel> = emptyList(),
    val totalAmount: BigDecimal = items.fold(BigDecimal.ZERO) { acc, item ->
        acc + (item.price * BigDecimal(item.quantity))
    }
)
